tools/makerootfs.sh

```diff
@@ -1,30 +1,116 @@
 #!/bin/bash
 # Usage:
 #
-#     ./makerootfs.sh <image.yml> <output rootfs image> <fs> <arch>
+#     ./makerootfs.sh mode [-y <image.yml>] [-i <output rootfs image>] [-t <tar file>] [-f <filesystem format>] [-a <arch>]
 # <fs> defaults to squash
 # <arch> defaults to the current machine architecture
 
 set -e
 set -o pipefail
 
+# mode 3 - generate image from yml
+do_image() {
+  if [ -z "$ymlfile" -o -z "$imgfile" -o -n "$tarfile" ]; then
+    echo "must supply ymlfile and imgfile and not tarfile" >&2
+    help
+  fi
+  # did we specify an architecture?
+  ARCHARG=""
+  if [ -n "$arch" ]; then
+    ARCHARG="-arch ${arch}"
+  fi
+  : > "$IMAGE"
+  linuxkit build -docker ${ARCHARG} -o - "$ymlfile" | docker run -i --rm -v /dev:/dev --privileged -v "$IMAGE:/rootfs.img" "${MKROOTFS_TAG}"
```

Comment:
Please add the shellcheck disable SC2086 to avoid someone else later seeing the shellcheck complaint and suggestion and breaking the code. Same for the other SC2086 complaints.